security/pkg/nodeagent/caclient/client.go

```diff
@@ -26,19 +26,20 @@ import (
 	citadel "istio.io/istio/security/pkg/nodeagent/caclient/providers/citadel"
 	gca "istio.io/istio/security/pkg/nodeagent/caclient/providers/google"
 	vault "istio.io/istio/security/pkg/nodeagent/caclient/providers/vault"
+	"istio.io/pkg/env"
 	"istio.io/pkg/log"
 )
 
 const (
-	googleCAName = "GoogleCA"
-	citadelName  = "Citadel"
-	vaultCAName  = "VaultCA"
-	ns           = "istio-system"
-
+	googleCAName  = "GoogleCA"
+	citadelName   = "Citadel"
+	vaultCAName   = "VaultCA"
 	retryInterval = time.Second * 2
 	maxRetries    = 100
 )
 
+var namespace = env.RegisterStringVar("NAMESPACE", "istio-system", "namespace istio control plane run in").Get()
```

Comment:
`namespace istio control plane run in` is not accurate, this is the namespace Nodeagent is installed in. Pilot, etc can be installed other places